hooks/build

```
@@ -0,0 +1,449 @@
+
+pwd
+# Determine version of the Airflow from version.py
+export AIRFLOW_VERSION=$(cat airflow/version.py - << EOF | python
+print(version.replace("+",""))
+EOF
+)
+# Check if we are running in the CI environment
+# In case of CI build we are not building CI cache. We use it and
+# We also use CI deps as base of the main image
+CI=${CI:=false}
+
+if [[ "${CI}" == "true" ]]; then
+    NON_CI="false"
+else
+    NON_CI="true"
+fi
+
+# Extras used to to build main airflow image
+AIRFLOW_MAIN_EXTRAS=${AIRFLOW_MAIN_EXTRAS:="all"}
+
+# Extras used to build cache and CI image
+AIRFLOW_CI_EXTRAS=${AIRFLOW_CI_EXTRAS:="devel_ci"}
+
+echo
+echo "Airflow ${AIRFLOW_VERSION} Python: ${PYTHON_VERSION}."
+echo
+
+# Whether to push images after build
+# This is set to false on CI builds
+export AIRFLOW_CONTAINER_PUSH_IMAGES=${AIRFLOW_CONTAINER_PUSH_IMAGES:=${NON_CI}}
+
+# Whether to force pull images to populate cache
+export AIRFLOW_CONTAINER_FORCE_PULL_IMAGES=${AIRFLOW_CONTAINER_FORCE_PULL_IMAGES:="true"}
+
+# Skips downloading and building the main, trimmed down image of airflow
+# This is set to true by default in CI environment (we only need CI image then)
+export AIRFLOW_CONTAINER_SKIP_MAIN_IMAGE=${AIRFLOW_CONTAINER_SKIP_MAIN_IMAGE:=${CI}}
+echo "Skip main image: ${AIRFLOW_CONTAINER_SKIP_MAIN_IMAGE}"
+
+# Skips pulling the airflow images - this will use cache but will build it all from scratch
+export AIRFLOW_CONTAINER_SKIP_PULLING_AIRFLOW_IMAGES=${AIRFLOW_CONTAINER_SKIP_PULLING_AIRFLOW_IMAGES:="false"}
+echo "Skip pulling Airflow images: ${AIRFLOW_CONTAINER_SKIP_PULLING_AIRFLOW_IMAGES}"
+
+export AIRFLOW_FIX_PERMISSIONS=${AIRFLOW_FIX_PERMISSIONS:="all"}
+echo "Fixing permissions: ${AIRFLOW_FIX_PERMISSIONS}"
+
+# Base python image for the build
+PYTHON_BASE_IMAGE=python:${PYTHON_VERSION}-slim
+
+# Image of the main airflow - this is a "reference" image of Airflow with minimum requirements needed
+AIRFLOW_IMAGE="${LOCAL_IMAGE}-v${AIRFLOW_VERSION}"
+
+# Image of the Airflow CI - this is the image used to run tests in CI environment
+AIRFLOW_CI_IMAGE="${LOCAL_IMAGE}-ci-v${AIRFLOW_VERSION}"
+
+# In the future we can enable buildkit.
+# It's experimental now and cache does not work out of the box with buildkit in Docker 18.09.2, buildkit 0.3.3
+# It is fixed in upcoming buildkit 0.4.0.
+# Buildkit will make build faster (including parallel builds of multi-stage builds).
+# It will also help with simpler skipping of unused images.
+export DOCKER_BUILDKIT=${DOCKER_BUILDKIT:=0}
+
+# List of images to push at the end of the build
+IMAGES_BUILT=""
+
+end_step
+
+AIRFLOW_CONTAINER_CLEANUP_IMAGES=${AIRFLOW_CONTAINER_CLEANUP_IMAGES:="false"}
+if [[ "${AIRFLOW_CONTAINER_CLEANUP_IMAGES}" == "true" ]]; then
+    echo
+    ${AIRFLOW_ROOT}/confirm "Removing all local images for that build."
+    echo
+    start_step "Removing images"
+    docker rmi ${PYTHON_BASE_IMAGE} || true
+    docker rmi ${AIRFLOW_IMAGE} || true
+    docker rmi ${AIRFLOW_CI_IMAGE} || true
```

```suggestion
    docker rmi "${AIRFLOW_CI_IMAGE}" || true
```
Double quote to prevent globbing and word splitting.

---

Done